opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -198,38 +217,75 @@ public List<DateFormatter> getTimeNamedFormatters() {
     return getFormatList().stream()
         .filter(formatString -> {
           FormatNames namedFormat = FormatNames.forName(formatString);
-          return SUPPORTED_NAMED_TIME_FORMATS.contains(namedFormat);
+          return namedFormat == null ? false : SUPPORTED_NAMED_TIME_FORMATS.contains(namedFormat);
         })
         .map(DateFormatter::forPattern).collect(Collectors.toList());
   }
 
-  private ExprCoreType getExprTypeFromFormatString(String formatString) {
-    if (formatString.isEmpty()) {
-      // FOLLOW-UP: check the default formatter - and set it here instead
-      // of assuming that the default is always a timestamp
-      return TIMESTAMP;
+  private ExprCoreType getExprTypeFromCustomFormatString(List<DateFormatter> formatters) {
+    // Characters from
+    // https://opensearch.org/docs/latest/field-types/supported-field-types/date/#built-in-formats
+    Pattern timeFormatChars = Pattern.compile("[HmsSZ]");
+    Pattern dateFormatChars = Pattern.compile("[yYMwdDe]");
+    boolean isTime = false;
+    boolean isDate = false;
+
+    for (DateFormatter formatter: formatters) {
+      if (timeFormatChars.matcher(formatter.pattern()).find()) {
+        isTime = true;
+      }
+      if (dateFormatChars.matcher(formatter.pattern()).find()) {
+        isDate = true;
+      }
+      if ((isDate && isTime)) {
+        return TIMESTAMP;
+      }
+    }
+    if (isTime) {
+      return TIME;
+    }
+    if (isDate) {
+      return DATE;
     }
 
-    List<DateFormatter> namedFormatters = getAllNamedFormatters();
+    // Default type
+    return TIMESTAMP;
```

Comment:
This line is unreached when all formatters contains no pre-defined letters. Is it possible?
Maybe you have to throw an exception here.

Reply:
Out of scope of the spike and will be part of the full implementation PR. Thanks for the comment.